client/analytics/components/report-table/index.js

```diff
@@ -84,15 +84,13 @@ class ReportTable extends Component {
 		}
 
 		const isRequesting = tableData.isRequesting || primaryData.isRequesting;
-		const orderedItems = orderBy( items.data, query.orderby, query.order );
```

Comment:
This logic is removed for two reasons:

* Ordering is already properly handled in the REST API.
* This won't correctly sort strings alphabetically.